### DISCUSSION TITLE
【SendMessage】Sending messages, occasionally 1-2 messages are sent with a delay of about 1s.

### DISCUSSION THREAD

**Discussion by humkum** [apache/rocketmq#7554] [Q&A] [answered]
![image](https://github.com/apache/rocketmq/assets/50660789/7a36acef-ed7a-4df7-9548-0a290fb44e9c)
Powerful community, have you ever encountered the following problems?
The cluster is deployed in DLedger mode (raft). Occasionally, it takes about 1 second for the cluster to send 1-2 messages. It is strange. In theory, the long time should be due to the length of a batch of messages except only one message！ There are currently no troubleshooting ideas. Do you have the same problem in master-slave mode? If so, what is the cause and how can it be solved?
万能的社区，大家是否遇到过这个问题？
集群是采用 DLedger  模式(raft)部署的，集群发送偶有 1-2 条耗时到 1s 左右，很奇怪，理论上耗时长应该是一批消息都长才对而不是单条耗时长。目前没有排查思路。请问主从模式下有相同的问题吗？如果有的话，请问是什么原因造成的，可以通过什么方式解决？

**Comment by francisoliverlee** [comment]
检查下延迟1-2s的那条消息是延迟消息嘛？ 在property中有延迟的标记。

**Reply by humkum** [reply]
不是延时消息

**Comment by cserwen** [comment, marked as answer]
已定位到，DLedger 低版本主从同步会有1s的毛刺情况，高版本（0.2.6以上）可能修复了，这个没有验证过，具体问题参考 https://github.com/openmessaging/dledger/issues/57
